hummingbot/connector/gateway_EVM_AMM.py

```diff
@@ -503,30 +503,26 @@ async def _update_token_approval_status(self, tracked_approvals: List[GatewayInF
                 self.logger().error(f"Error while trying to approve token {token_symbol} for {self.connector_name}: "
                                     "txHash key not found in transaction status.")
                 continue
-            if transaction_status["txStatus"] == 1:
-                if transaction_status["txReceipt"]["status"] == 1:
-                    self.logger().info(f"Token approval for {tracked_approval.client_order_id} on {self.connector_name} "
-                                       f"successful.")
-                    self.trigger_event(
-                        TokenApprovalEvent.ApprovalSuccessful,
-                        TokenApprovalSuccessEvent(
-                            self.current_timestamp,
-                            self.connector_name,
-                            token_symbol
-                        )
+            if transaction_status["txStatus"] in [-1, 1]:
```

Comment:
Same here. We should not treat txStatus -1 as failure - it simply hasn't been mined yet. This will cause pretty much all transactions to appear to fail.